File: C/Array & String/334. Increasing Triplet Subsequence.c
/* 334. Increasing Triplet Subsequence
Difficulty: Medium
Submission Link: https://leetcode.com/problems/increasing-triplet-subsequence/submissions/1030280770/
 */

bool increasingTriplet(int* nums, int numsSize) {
    int min = INT_MAX;
    int mid = INT_MAX;

    for (int i = 0; i < numsSize; i++) {
        if (nums[i] <= min) {
            min = nums[i];
        } else if (nums[i] <= mid) {
            mid = nums[i];
        } else {
            return true;
        }
    }

    return false;
}

File: C#/Array & String/334. Increasing Triplet Subsequence.cs
/* 334. Increasing Triplet Subsequence
Difficulty: Medium
https://leetcode.com/problems/increasing-triplet-subsequence/submissions/1030280898/
 */

 public class Solution {
    public bool IncreasingTriplet(int[] nums) {

        if(nums.Length < 3) return false;

        int smallest = int.MaxValue;
        int secondSmallest = int.MaxValue;

        for(int idx = 0; idx < nums.Length; idx++)
        {
            if(nums[idx] <= smallest)
                smallest = nums[idx];
            else if(nums[idx] <= secondSmallest)
                secondSmallest = nums[idx];
            else 
                return true;
        }
        return false;
    }
}